src/Relude/Nub.hs

```diff
@@ -90,3 +97,27 @@ sortNub = Set.toList . Set.fromList
 -}
 unstableNub :: (Eq a, Hashable a) => [a] -> [a]
 unstableNub = HashSet.toList . HashSet.fromList
+
+{- | A generalized nub for any 'Foldable'. Not necessarily order preserving. -}
+foldNub :: (Foldable t, Ord a) => t a -> Set.Set a
+foldNub = foldr Set.insert Set.empty
+
+{- | A strict variant of 'foldNub' using 'foldl'' instead of 'foldr'-}
+foldNub' :: (Foldable t, Ord a) => t a -> Set.Set a
+foldNub' = foldl' (flip Set.insert) Set.empty
```

Comment:
I think we don't need to have `foldNub` function and having `foldNub'` only and renaming `foldNub'` to `foldNub` is okay. Because of the `foldr` implementation, `foldr` is better than `foldl'` only the given function is left-lazy, i.e. it can stop evaluating by looking at the current element of the `Foldable` and it evaluates left argument before evaluating right. For example, `foldr (&&)` is an okay use-case. But this doesn't work for `Set.insert` because you always evaluate the whole `Foldable` and using `foldr` will result in decreased performance because it will keep the whole evaluation stack in memory before finishing calculation. So using `foldl'` is the only correct choice.

Reply:
Thanks! I'll change this for `foldNub`. This applies to `hashFoldNub` also, right?

Reply:
Yes, same for `hasFoldNub` as well.